src/Approximations/overapproximate.jl

```diff
@@ -1510,3 +1510,123 @@ function overapproximate(am::AbstractAffineMap{N, <:AbstractHyperrectangle{N}},
+Under this constraint, the zonotope has the minimal sum of generator vectors.
+
+### Notes
+
+The algorithm only requires one representative of each generator direction and
+their additive inverse (e.g. only one of `[1, 0]` and `[-1, 0]`) and assumes
+that the directions are normalized.
+We preprocess the directions in that respect.
+
+### Algorithm
+
+We solve a linear program parametric in the vertices ``v_j`` of `X` and the
+directions ``d_k`` in `dir` presented in [1, Section 4.2] (adapting the notation
+to the one used in this library).
+
+```math
+    \\min \\sum_{k=1}^l α_k \\
+    s.t. \\
+    c + \\sum_{k=1}^l b_{kj} * d_k = v_j \\quad \\forall j \\
+    -α_k ≤ b_{kj} ≤ α_k \\quad \\forall k, j \\
+    α_k ≥ 0 \\quad \\forall k
+```
+
+The resulting zonotope has center `c` and generators `α_k · d_k`.
+
+Note that the first type of side constraints is vector-based and that the
+nonnegativity constraints (last type) are not stated explicitly in [1].
+
+[1] Zonotopes as bounding volumes, L. J. Guibas et al, Proc. of Symposium on
+    Discrete Algorithms, pp. 803-812.
+"""
+# note: this set can generally be reduced
+function overapproximate(X::LazySet{N}, ::Type{<:Zonotope},
+                         dir::AbstractDirections{N};
+                         solver=default_lp_solver(N)) where {N<:Real}
+    function normalize_directions(dir::AD) where {AD}
+        dirs = Vector{Vector{N}}()
+        for d in dir
+            d_normalized = normalize(d)
+            if (d_normalized ∉ dirs) && (-d_normalized ∉ dirs)
+                push!(dirs, d_normalized)
+            end
+        end
+        return dirs
+    end
+
+    n = dim(X)
+    dirs = normalize_directions(dir)
+    l = length(dirs)
+    V = vertices_list(X)
+    m = length(V)
+    nvariables = l + n + l * m  # l 'α_k' + n 'p[i]' + lm 'b_lj'
+    nconstraints = n * m + 2 * l * m  # nm 'p_j' + 2lm 'b_kj'
+
+    obj = vcat(ones(l), zeros(nvariables - l))
+    A = spzeros(N, nconstraints, nvariables)
+    sense = vcat(fill('=', n * m), fill('<', 2 * l * m))
+    b = zeros(N, nconstraints)
+    r = 1
+    # constraints p[i] + \sum_k v_j*b_kj = p_j
+    col_offset_p = l
+    col_offset_b = l + n + 1
+    for (j, vj) in enumerate(V)
+        for i in 1:n
+            A[r, col_offset_p + i] = N(1)  # p[i]
+            for (k, dk) in enumerate(dirs)
+                A[r, col_offset_b + (k - 1) * m] = dk[i]
+            end
+            b[r] = vj[i]
+            r += 1
+        end
+        col_offset_b += 1
+    end
+    # constraints -α_k ± b_kj <= 0
+    col_offset_b = l + n
+    for k in 1:l
+        for j in 1:m
+            for sign_b in N[1, -1]
+                A[r, k] = N(-1)  # - α_k
+                A[r, col_offset_b + j] = sign_b  # ± b_kj
+                r += 1
+            end
+        end
+        col_offset_b += m
+    end
+    lower_bounds = vcat(zeros(N, l), fill(N(-Inf), nvariables - l))
+    lp = linprog(obj, A, sense, b, lower_bounds, Inf, solver)
```

Comment:
for the problem formulation, i would prefer that we use the JuMP interface which is much less verbose (dependency which is anyways loaded)

Reply:
I tried that first and spent quite some time but could not get it to work. All other LPs in the library also do not use `JuMP` so I do not see why this is necessary now.